src/gui_common/GUICommon.cs

```diff
@@ -90,12 +90,18 @@ public void ModulateFadeIn(Control control, float duration)
         tween.Start();
     }
 
-    public void ModulateFadeOut(Control control, float duration)
+    public async void ModulateFadeOut(Control control, float duration, bool hideOnFinished = true)
     {
         control.Modulate = new Color(1, 1, 1, 1);
 
         tween.InterpolateProperty(control, "modulate:a", 1, 0, duration, Tween.TransitionType.Sine, Tween.EaseType.In);
         tween.Start();
+
+        if (hideOnFinished)
+        {
+            await ToSignal(tween, "tween_completed");
```

Comment:
First time I'm seeing this kind of code. Also my IDE is complaining (though the actual build seems to succeed):
  <Thrive>\src\gui_common\GUICommon.cs:2828 Type 'Godot.SignalAwaiter' is not awaitable

Reply:
This is like the equivalent of connecting a signal to a callback method, except this one is more cleaner and less code to write. And documented as C# equivalent of GDScript's `yield` https://docs.godotengine.org/en/stable/getting_started/scripting/c_sharp/c_sharp_differences.html#yield. Since this doesn't cause problem, perhaps the warning could be suppresed?

Reply:
I can't seem to find anything, about suppressing it. I think the problem might be that Godot.IAwaiter doesn't implement `ICriticalNotifyCompletion` interface, so this might be a Godot bug? This is what I used as reference for how awaitable types are made: https://putridparrot.com/blog/creating-awaitable-types/ Perhaps it's possible to make a correctly constructed Awaitable wrapper that could be used here?
Once this is done: https://github.com/Revolutionary-Games/Thrive/issues/2058 CI might also start complaining about this...
so some kind of solution is needed here

Reply:
Will be hard to confirm any fix done on this since I can't replicate the build warnings, I guess I'll opt with the usual way of handling signals in the meantime.

Reply:
Sure, that's fine. I'll open an issue for reviewing why exactly `await ToSignal(tween, "tween_completed");` triggers the problem.